cmd/gitserver/server/vcs_syncer_perforce.go

```diff
@@ -39,13 +66,39 @@ func (s *PerforceDepotSyncer) Type() string {
 
 // IsCloneable checks to see if the Perforce remote URL is cloneable.
 func (s *PerforceDepotSyncer) IsCloneable(ctx context.Context, remoteURL *vcs.URL) error {
-	username, password, host, _, err := decomposePerforceRemoteURL(remoteURL)
+	username, password, host, path, err := decomposePerforceRemoteURL(remoteURL)
 	if err != nil {
 		return errors.Wrap(err, "decompose")
 	}
 
-	// FIXME: Need to find a way to determine if depot exists instead of a general ping to the Perforce server.
-	return p4pingWithTrust(ctx, host, username, password)
+	// start with a test and set up trust if necessary
+	if err := p4testWithTrust(ctx, host, username, password); err != nil {
+		return err
+	}
+
+	// the path could be a path into a depot, or it could be just a depot
+	// expect it to start with at least one slash
+	// (the config defines it as starting with two, but converting it to a URL may change that)
+	// the first path part will be the depot - subsequent parts define a directory path into a depot
+	// ignore the directory parts for now, and only test for access to the depot
+	// TODO: revisit if we want to also test for access to the directories, if any are included
+	depot := strings.Split(strings.TrimLeft(path, "/"), "/")[0]
+
+	// get a list of depots that match the supplied depot (if it's defined)
+	if depots, err := p4depots(ctx, host, username, password, depot); err != nil {
+		return err
+	} else if len(depots) == 0 {
```

Comment:
More idiomatic, kinda :)

```suggestion
	}
	if len(depots) == 0 {
```

Reply:
Actually, in this case, that broke the build because it pulled the reference to `depots` out of scope: `depots` is set inside the `if` block, so it can't be referenced outside of it.